lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -966,12 +1024,14 @@ void LowerClassesPass::lowerClass(om::ClassOp classOp, FModuleLike moduleLike,
     if (!outputPort)
       continue;
 
-    // Get the original port name, create a Field, and erase the propassign.
-    auto name = moduleLike.getPortName(outputPort.getArgNumber());
-    builder.create<ClassFieldOp>(op.getLoc(), name, op.getSrc());
+    locs.push_back(op.getLoc());
+    fieldValues.push_back(op.getSrc());
+
     op.erase();
   }
```

Comment:
We now have to traverse the class definition and collect PropAssignOps twice, once to make the field/type dict, another here to make the fields terminator.  If we wanted to avoid this, we could try to collect the values for the field terminator earlier (when creating the containing class op) and just forward that info here to create the terminator op rather than doing a second traversal.

Reply:
> we could try to collect the values for the field terminator earlier (when creating the containing class op) and just forward that info here to create the terminator op rather than doing a second traversal

This might be worth it... we have tried to optimize this pass, since it can be quite expensive, so if you already know you're doing two passes, it might be worth making one pass. I think in general, we should take a look at the performance before and after this PR regardless, and it make sense to spend some time on optimizations like this.

Reply:
Looking at `LowerClassesPass::runOnOperation` I think it should be straightforward to pass the information through so I'll take a look at it.

Reply:
I was able to "fuse" the loop with the previous loop, so I think this PR includes net the same amount of traversals now.  See https://github.com/llvm/circt/pull/7537/commits/945e497518526065748e17533dd1342b5ef3d791

We could further optimize this by traversing the ops once, collecting the relevant information but delaying construction of the class body until afterwards.  The main issue is that we need to traverse to construct the fieldTypes map before traversing to collect the class body.  I'd imagine the performance improvements here would be marginal if any (or worse) since we'd essentially need to do a similar complexity traversal of the stored information.  So I feel we can hold off on that until we do some profiling and decide it's a worthy place to improve

Reply:
Makes sense, that commit looks good.